flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/JobManagerOperatorGroupTest.java

```diff
@@ -0,0 +1,226 @@
+
+    @Test
+    void testGenerateScopeDefault() {
+        final JobID jobId = new JobID();
+        final JobVertexID jobVertexId = new JobVertexID();
+        final OperatorID operatorId = new OperatorID();
+        JobManagerOperatorMetricGroup jmJobGroup =
+                JobManagerMetricGroup.createJobManagerMetricGroup(registry, "localhost")
+                        .addJob(jobId, "myJobName")
+                        .getOrAddOperator(jobVertexId, "taskName", operatorId, "opName");
+
+        assertThat(jmJobGroup.getScopeComponents())
+                .containsExactly("localhost", "jobmanager", "myJobName", "opName");
+        assertThat(jmJobGroup.getMetricIdentifier("name"))
+                .isEqualTo("localhost.jobmanager.myJobName.opName.name");
+    }
+
+    @Test
+    void testGenerateScopeCustom() throws Exception {
+        Configuration cfg = new Configuration();
+        cfg.setString(
+                MetricOptions.SCOPE_NAMING_JM_OPERATOR,
+                "constant.<host>.foo.<host>.<job_id>.<job_name>.<task_id>.<task_name>.<operator_id>.<operator_name>");
+        MetricRegistryImpl registry =
+                new MetricRegistryImpl(MetricRegistryTestUtils.fromConfiguration(cfg));
+
+        final JobID jobId = new JobID();
+        final JobVertexID jobVertexId = new JobVertexID();
+        final OperatorID operatorId = new OperatorID();
+        JobManagerOperatorMetricGroup jmJobGroup =
+                JobManagerMetricGroup.createJobManagerMetricGroup(registry, "host")
+                        .addJob(jobId, "myJobName")
+                        .getOrAddOperator(jobVertexId, "taskName", operatorId, "opName");
+
+        assertThat(jmJobGroup.getScopeComponents())
+                .containsExactly(
+                        "constant",
+                        "host",
+                        "foo",
+                        "host",
+                        jobId.toString(),
+                        "myJobName",
+                        jobVertexId.toString(),
+                        "taskName",
+                        operatorId.toString(),
+                        "opName");
+        assertThat(jmJobGroup.getMetricIdentifier("name"))
+                .isEqualTo(
+                        String.format(
+                                "constant.host.foo.host.%s.myJobName.%s.taskName.%s.opName.name",
+                                jobId, jobVertexId, operatorId));
+
+        registry.closeAsync().get();
+    }
+
+    @Test
+    void testGenerateScopeCustomWildcard() throws Exception {
+        Configuration cfg = new Configuration();
+        cfg.setString(MetricOptions.SCOPE_NAMING_JM, "peter");
+        cfg.setString(MetricOptions.SCOPE_NAMING_JM_JOB, "*.some-constant.<job_id>");
+        cfg.setString(MetricOptions.SCOPE_NAMING_JM_OPERATOR, "*.other-constant.<operator_id>");
+        MetricRegistryImpl registry =
+                new MetricRegistryImpl(MetricRegistryTestUtils.fromConfiguration(cfg));
```

Comment:
Use a `TestingMetricRegistry` instead. This is cheaper and avoids the need for cleaning up any resources. Also applies to other _added_ tests.